### PR TITLE
Feature/vethub 265

### PR DESCRIPTION
**Pull request by Jo-Zh**
implement receive user Attempts associated Smart Quiz
- to test 'yarn test', should not broken existed tests.  **note** the algorithm test haven't been created since it hasn't been implemented.
- to verify in `test sso` refer to: https://github.com/UQ-eLIPSE/crucible/pull/172

### CODE REVIEW

**Comment by freshc0w** [comment]
One test failed
image

File: src/components/QuestionStore.ts [hunk 1]
```diff
@@ -13,12 +13,24 @@ export const shuffleArray = (array: MCQuestion[]) => {
...
 };
 
+export const srsSortedQuestions = (questions: MCQuestion[]) => {
+  return questions.sort(
+    (curr, next) =>
+      curr.lastAttempted > next.lastAttempted
+        ? 1
+        : curr.lastAttempted > next.lastAttempted
+          ? -1
+          : 0, //todo:replace this with reviewTime once algorithm implemented
```
Comment (freshc0w) [inline src/components/QuestionStore.ts]: The conditions are repeated ?

Comment (freshc0w) [reply src/components/QuestionStore.ts]: I saw the TODO. Does this mean line 21 will be replaced by `curr.reviewTime > next.reviewTime` in the future?

Comment (Jo-Zh) [reply src/components/QuestionStore.ts]: > I saw the TODO. Does this mean line 21 will be replaced by `curr.reviewTime > next.reviewTime` in the future?

Yes, it going to be replaced in the Ticket 266 with SRS algorithm, but the repeat is a typo will fix it.  We can just simply implement as see what is the feedback.  I put this function temporily here, in order to not broken the app and indicate where to place the algorithm and

**Review by anzau — APPROVED** [review]

**Pull request merged**